template/template.go

```diff
@@ -103,6 +105,15 @@ func WithSelectKey(selectKey func(keys []string) string) ResolvedValueOpt {
 	}
 }
 
+// WithFunction applies a function name and options with which that function was called to the
+// [ResolvedValue]. It allows consecutive functions of the same type to preserve and re-use options.
+func WithFunction(funcName string, opts Options) ResolvedValueOpt {
```

Comment:
Let's not expose this function, it has no practical use to others who want to register a new function.

```suggestion
func withFunction(funcName string, opts Options) ResolvedValueOpt {
```